test/e2e/specs/editor/blocks/comments.spec.js

```diff
@@ -193,4 +371,48 @@ class CommentsBlockUtils {
 
 		return previousValue;
 	}
+
+	async hideWelcomeGuide() {
+		await this.page.evaluate( async () => {
+			const isWelcomeGuideActive = window.wp.data
+				.select( 'core/edit-post' )
+				.isFeatureActive( 'welcomeGuide' );
+
+			if ( isWelcomeGuideActive ) {
+				window.wp.data
+					.dispatch( 'core/edit-post' )
+					.toggleFeature( 'welcomeGuide' );
+			}
+		} );
+
+		await this.page.reload();
+		await this.page.waitForSelector( '.edit-post-layout' );
+	}
+}
+
+class CodeEditorUtils {
+	constructor( { page } ) {
+		this.page = page;
+	}
+
+	async open() {
+		await this.page
+			.locator(
+				'role=region[name="Editor top bar"i] >> role=button[name="Options"i]'
+			)
+			.click();
+		await this.page
+			.locator( 'role=menuitemradio[name*="Code editor"i]' )
+			.click();
+	}
+
+	textbox() {
+		return this.page.locator( 'role=textbox[name="Type text or HTML"i]' );
+	}
+
+	async close() {
+		await this.page
+			.locator( 'role=button[name="Exit code editor"i]' )
+			.click();
+	}
```

Comment:
This might actually come in handy as a e2e util for other tests. (Not a blocker, can do in a follow-up PR.)
Curious to hear @kevin940726's thoughts.

Reply:
I thought the same. It could be part of the [`editor` utils](https://github.com/WordPress/gutenberg/blob/65a89c9fbf7a460838b5fb9080605b05c0c2b698/packages/e2e-test-utils-playwright/src/editor/index.ts). 🙂

Reply:
Normally, we'll like to minimize the API interface of the utils. As per the [best practices](https://github.com/WordPress/gutenberg/blob/trunk/test/e2e/README.md#dont-overload-test-utils-inline-simple-utils), inlining simple utils is often more readable and easier to be discovered. For this example, we can simplify it a little bit by directly calling the `.click( selector )` API instead (`.locator( selector ).click()` is the same as `.click( selector )`). Then the close function will be able to fit in a single line. With the help of role selectors, the line should be fairly readable without having to jump around the source code.

If however, we want only to get the serialized HTML of the blocks, then maybe `getEditedPostContent()` is enough? If the test is not about the code editor, then we should be able just to use the public or private API to do most of the job instead.

Reply:
Yup, `getEditedPostContent()` is exactly what it's needed there. Thanks for your help @kevin940726! 🙌

Reply:
Thanks for your advice @kevin940726, insightful as always! 😄